### DISCUSSION TITLE
Copilot-chat extra does not work with Blink extra enabled

### DISCUSSION THREAD

**Discussion by tkivela** [LazyVim/LazyVim#4711] [Q&A] [answered]
Hello,
Enabled Blink extra yesterday and it seems it doesn't play together with Copilot-chat extra. Problem seems to be the hardcoded line in the copilot-chat extra which refers to cmp:
```lua
require("CopilotChat.integrations.cmp").setup()
```
This hardcoding prevents copilot-chat opening at all if blink extra is enabled.

Has anyone else experienced this?

**Comment by tkivela** [comment]
Same thing seems to be reported in CopilotC repo in this issue https://github.com/CopilotC-Nvim/CopilotChat.nvim/issues/429, but no issue about it seems to be opened in Lazyvim repo

**Reply by dpetka2001** [reply, marked as answer]
`blink.cmp` is still a new Extra and not yet fully compatible with all AI Extras. I believe you could solve it by doing 
```lua
    config = function(_, opts)
      local chat = require("CopilotChat")
      if LazyVim.has("nvim-cmp") then
        require("CopilotChat.integrations.cmp").setup()
      end

      vim.api.nvim_create_autocmd("BufEnter", {
        pattern = "copilot-chat",
        callback = function()
          vim.opt_local.relativenumber = false
          vim.opt_local.number = false
        end,
      })

      chat.setup(opts)
    end,
```
I don't use Copilot, so can't test it properly, but if you manage to solve this with the above code snippet, feel free to submit a PR to rectify this behavior.
